src/game/ChoiceHandling.hpp

```diff
@@ -99,22 +101,40 @@ namespace actions {
             const std::vector<spy::character::CharacterInformation> &charInfos =
                     root_machine(fsm).characterInformations;
             spy::gameplay::State &gameState = root_machine(fsm).gameState;
+            const ChoiceSet &choiceSet = root_machine(fsm).choiceSet;
+            const unsigned int maxNumberOfNPCs = root_machine(fsm).maxNumberOfNPCs;
 
             auto charsP1 = s.characterChoices.at(playerIds.at(Player::one));
             auto charsP2 = s.characterChoices.at(playerIds.at(Player::two));
 
             spy::character::CharacterSet charSet;
 
             spy::character::FactionEnum faction;
-            //TODO: check if all non-chosen characters are NPCs
+
+            auto remainingCharacters = choiceSet.getRemainingCharacters();
+            auto remainingGadgets    = choiceSet.getRemainingGadgets();
+
+            std::list<spy::util::UUID> npcCharacters;
+            // choose characters that will be NPCs
+            if (remainingCharacters.size() <= maxNumberOfNPCs) {
+                npcCharacters = remainingCharacters;
+            } else {
+                for (auto i = 0U; i < maxNumberOfNPCs; i++) {
+                    auto uuidIterator = spy::util::GameLogicUtils::getRandomItemFromContainer(remainingCharacters);
+                    npcCharacters.push_back(*uuidIterator);
+                    remainingCharacters.erase(uuidIterator);
+                }
+            }
+
             for (const auto &c : charInfos) {
                 if (std::find(charsP1.begin(), charsP1.end(), c.getCharacterId()) != charsP1.end()) {
                     faction = spy::character::FactionEnum::PLAYER1;
                 } else if (std::find(charsP2.begin(), charsP2.end(), c.getCharacterId()) != charsP2.end()) {
                     faction = spy::character::FactionEnum::PLAYER2;
-                } else {
-                    // not chosen -> NPC
+                } else if (std::find(npcCharacters.begin(), npcCharacters.end(), c.getCharacterId()) != npcCharacters.end()) {
```

Comment:
syntax mit `find != end` ist so ein bisschen unleserlich. Vielleicht Kommentar über das if, z.B.  
`// search for the character in the factions, and assign the correct faction`

und vielleicht über die for-schleife:  
`// for each character, insert it into the character set if it has been assigned to a faction (or NPC)`